src/components/app.tsx

```diff
@@ -667,7 +671,11 @@ export class App extends React.Component<AppProps, AppState> {
         if (sensorManager) {
             sensorManager.requestSleep();
             // pause before attempting to reload SensorConnector
-            setTimeout(() => sensorManager.requestWake(), SLEEP_WAKE_DELAY_SEC * 1000);
+            setTimeout(() => {
+                if (sensorManager) {
```

Comment:
I think it's necessary to test `this.state.sensorManager` here, as the `sensorManager` variable was captured at the time of the closure and so can't become null during the period of the timeout. The case we're trying to catch is one in which the user disconnects the sensorManager, which should result in resetting `this.state.sensorManager`.

Reply:
Ah, I see what you're saying.  Sounds correct and safer to me.  I changed the test to use what is actually in `this.state.sensorManager` instead of the `sensorManager` variable that was set before the timer fired.